.eslintrc.json

```diff
@@ -18,6 +18,8 @@
       "import/no-unresolved": "off",
       "no-shadow": "off",
       "arrow-parens": "off",
-      "react/jsx-props-no-spreading": "off"
+      "react/jsx-props-no-spreading": "off",
+      "no-param-reassign": "off",
+      "import/extensions": "off"
```

Comment:
**"react/jsx-props-no-spreading": "off",**
Is this a hard requirement for the project?
![image](https://user-images.githubusercontent.com/9512604/72151202-6c126300-3386-11ea-8d85-3e37bfbcf268.png)
This kind of contructions using spread operator and short circuit are recommended by many sources.
https://www.techiediaries.com/react-spread-operator-props-setstate/

**"no-param-reassign": "off",**
The own project requirements instruct us to mutate the *calculator data object*, this would not be possible with the above rule turned on.

"import/extensions": "off"
The NPM package *big.js* will fail the import since its own name finish with *.js*, I have googled for a workaround but this is by far the cleaner solution.

Reply:
> ### "react/jsx-props-no-spreading": "off"

I don't think this it is needed since:
1. it seems to be used only for this component and 
2. it can be easily readable by explicitly stating the props.

- [Current code]( https://github.com/ferreirati/mv-react-01-calculator/blob/56d4baa97e1b63631560d161ef6dc6e8cd5dfb41/src/components/ButtonPanel.js)
```jsx
      <div className="button-panel">
        {buttons.map((row, i) => (
          <div key={row[0]}>
            {row.map((b, j) => (
              <Button
                key={b}
                name={b}
                {...(b === '0' && { wide: true })}
                {...(i < 4 && j < 3 && { color: 'lightgrey' })}
                {...(i === 4 && j < 2 && { color: 'lightgrey' })}
                clickHandler={this.handleClick}
              />
            ))}
          </div>
        ))}
      </div>
```

- Modified code

```jsx
      <div className="button-panel">
        {buttons.map((row, i) => (
          <div key={row[0]}>
            {row.map((b, j) => (
              <Button
                key={b}
                name={b}
                wide={b === '0'}
                color={j < row.length - 1 ? 'lightgrey' : undefined}
                clickHandler={this.handleClick}
              />
            ))}
          </div>
        ))}
      </div>
```

You can also rename your variables into more meaningful names. 


> #### "no-param-reassign": "off",
The own project requirements instruct us to mutate the calculator data object, this would not be possible with the above rule turned on.

I think you miss the context of the instruction also you don't need to do much to fix this see the diff below for the [calculate function](https://github.com/ferreirati/mv-react-01-calculator/blob/56d4baa97e1b63631560d161ef6dc6e8cd5dfb41/src/logic/calculate.js):

```diff
-  const calculate = ({ total, next, operation }, buttonName) => {
+ const calculate = (data, buttonName) => {
  const operations = ['+', '-', 'x', '÷', '%'];

+  let { total, next, operation } = data;

```

> #### "import/extensions": "off"

I think you used the wrong package here. See https://www.npmjs.com/package/big-js, it `big-js` not `big.js`.


In summary, the main reason why we should not change correct file is to enforce consistence in our code and in code reviews and focus on more functional features on the code. If you still strongly think changing the config is necessary. I would encourage you to open a PR in the linter config repo.